extra_docs/relations/README.md

```diff
@@ -0,0 +1,163 @@
+# Relations
+All project resources can be related to each other and SDK supports inserting, fetching and attaching/detaching related 
+records.
+Let's create a simple schema which will be used for all further examples:
+
+![Relations schema](https://jexia.github.io/jexia-sdk-js/assets/relations-schema.png) 
+
+There are three datasets and two relations: **posts** has *one-2-many* relation to **comments** and **comments** has 
+*one-2-one* relation to **author** (remember that all management opearations, such as creating datasets, fields and 
+relations can be done with [Web Management Application](https://docs.jexia.com/getting-started/user-management/) only).
+
+To make our datasets not so boring, we can also add a few fields and create appropriate interfaces 
+for our typescript code:
+```typescript
+interface Author {
+  email: string;
+}
+interface Comment {
+  text: string;
+  like: boolean;
+  author: Author;
+}
+interface Post {
+  title: string;
+  message: string;
+  comments: Comment[];
+}
+```
+Pay your attention - **Post** includes **comments** field as an array, but **Comment** has an **author** 
+as a plain value. Thats the difference between *one-2-many* and *one-2-one* relation types.
+
+### [Insert data](#insert)
+To insert related records just provide a nested array to the `.insert()` method:
+```typescript
+dom.dataset("posts")
+  .insert([{
+    title: "Relations with Jexia SDK",
+    message: "A do like how Jexia SDK implements relations!",
+    comments: [{
+      text: "Me too!",
+      like: true,
+      author: { email: "customer@email.com" }
+    }, {
+      text: "Might've been a bit better :(",
+      like: false,
+      author: { email: "hater@yahoo.com" }
+    }]
+  }])
+  .execute();
+```
+
+Notice that records should be inserted in appropriate order - the relation root must be insert root. 
+It's impossible to insert into comment dataset with post related to that comment (tree cannot grow upside down). 
+Also look how we are using arrays for *one-2-many* **posts** -> **comments** relation and just an object 
```

Comment:
It is not obvious for me, can we add example, please?

Reply:
Ok, I will add some wrong examples :+1: